### PR TITLE
US115407 Initial Learning Outcomes integration

### PR DESCRIPTION
**Pull request by emil-furniss**
- new `d2l-activity-outcomes` component which can be integrated into a given activity editor. This will render the opener button if there are no alignments yet, or the tags for any existing aligned outcomes.
- we need to respect the lang term set by the config variable `d2l.Languages.Terminology.LearningOutcomes` in the LMS, which allows admins to set the terminology for learning outcomes: "Outcomes" vs "Standards" vs "Objectives" vs "Expectations" vs "Competencies". We do this via a prop that is set on the tag in the LMS renderer, and then access the term when required via the `requestProvider` event handler in `d2l-activity-assignment-editor.js`.
- conditional rendering within `d2l-activity-outcomes` is dependent on a couple of new props in the `ActivityUsage` mobx class: If `alignmentsHref` is defined on the activity usage entity, then outcomes are enabled in general. We fetch the `AlignmentsCollectionEntity` from that href, and that tells us if the user has permissions to modify the alignments (`canUpdateAlignments`) and which, if any, alignments are present (`hasAlignments`).

Required additions to the `siren-sdk`, incl new `AlignmentsCollectionEntity`:
https://github.com/BrightspaceHypermediaComponents/siren-sdk/pull/179/files

Passing config lang term to activities renderer in the LMS:
https://git.dev.d2l/projects/CORE/repos/lms/pull-requests/10546/diff

**EDIT:** This work needs to be flagged behind the M3 milestone, so the component now also checks the milestone via the `ActivityEditorFeaturesMixin`. I also added a `hidden` attribute to the component for when it is not being displayed, so we can add/hide the 20px padding in the parent component `AssignmentEditorDetail`. Previously the padding was always there, even if the outcomes component was not showing anything.

https://rally1.rallydev.com/#/29180338367d/detail/userstory/382756597864

### CODE REVIEW

File: components/d2l-activity-editor/d2l-activity-outcomes.js [hunk 1]
```diff
@@ -29,20 +32,34 @@ class ActivityOutcomes extends ActivityEditorMixin(LocalizeMixin(RtlMixin(MobxLi
...
 
 		this.addEventListener('d2l-alignment-list-added', this._onDialogAdd);
 		this.addEventListener('d2l-activity-alignment-outcomes-updated', this._onOutcomeTagDeleted);
 		this.addEventListener('d2l-alignment-list-cancelled', this._onDialogCancel);
 	}
 
+	disconnectedCallback() {
+		super.disconnectedCallback();
+
+		this.removeEventListener('d2l-alignment-list-added', this._onDialogAdd);
```
Comment (mdgbayly) [inline components/d2l-activity-editor/d2l-activity-outcomes.js:50]: nit: ? Are these events coming from the outcomes components? If so could we just add declarative event handlers to those components?

File: components/d2l-activity-editor/d2l-activity-outcomes.js [hunk 2]
```diff
@@ -99,10 +121,12 @@ class ActivityOutcomes extends ActivityEditorMixin(LocalizeMixin(RtlMixin(MobxLi
 		} = activity;
 
 		if (!canUpdateAlignments) {
+			this.hidden = true;
```
Comment (mdgbayly) [inline components/d2l-activity-editor/d2l-activity-outcomes.js:124]: Does rendering nothing still cause some space to be taken up in the host component? Just wondering why we need to explicitly hide this component?

File: components/d2l-activity-editor/d2l-activity-outcomes.js (hunk 1 above)
Comment (emil-furniss) [reply components/d2l-activity-editor/d2l-activity-outcomes.js:50]: yes they are, so yes we can! I'll move them over in the next PR.

File: components/d2l-activity-editor/d2l-activity-outcomes.js (hunk 2 above)
Comment (emil-furniss) [reply components/d2l-activity-editor/d2l-activity-outcomes.js:124]: No, it doesn't take up any space, but we want the parent to be able to apply padding when this component is displaying something. So this is just a way to signal that to the parent. I feel like there might be a much simpler way of doing that, but couldn't find it.

**Pull request merged**